share/spice/biblio_search/books_on_personality/biblio_search_books_on_personality.js

```diff
@@ -0,0 +1,85 @@
+(function(env) {
+  "use strict";
+
+  env.ddg_spice_biblio_search_books_on_personality = function(api_result) {
+    console.info(api_result);
+    
+    String.prototype.trunc = String.prototype.trunc ||
+      function(n){
+          return (this.length > n) ? this.substr(0,n-1)+'&hellip;' : this;
+      };
+    
+    // Validating response
+    if(!api_result || api_result.GoodreadsResponse.search["total-results"].text == "0") {
+      return Spice.failed('biblio_search_books_by_authors');
+    }
+    
+    var personality = api_result.GoodreadsResponse.search.query.text,
+      source_url = "https://www.goodreads.com/search?q=" + personality + "&search_type=books";
+    
+    var get_details_uri = '/js/spice/biblio_search/get_book_details/',
+      get_image_for_book_uri = '/js/spice/biblio_search/get_image_for_book/';
+    
+    // Render the response
+    Spice.add({
+      id: "biblio_search_books_on_personality",
+      
+      // Customize these properties
+      name: "Books",
+      data: api_result.GoodreadsResponse.search.results.work,
+      meta: {
+        primaryText: "Books on " + personality,
+        sourceName: "GoodReads",
+        sourceUrl: source_url
+      },
+      onItemShown: function(item) {
+        $.ajaxSetup({ cache: true });
+        
+        // Good reads description API
+        $.getJSON(get_details_uri + item.best_book.id.text, function(data) {
+          console.info(data);
+          
+          var description = data.GoodreadsResponse.book.description.text || "-- No Description Found --";
+          
+          // The desciption might contain some html tags. Removing those...
+          item['abstract'] = description && description.replace(/<\/?\w*>/gm, '').trunc(400);
+          item['heading'] = item.best_book.title.text;
+          item['isbn'] = data.GoodreadsResponse.book.isbn && data.GoodreadsResponse.book.isbn.text;
+          item['isbn13'] = data.GoodreadsResponse.book.isbn13 && data.GoodreadsResponse.book.isbn13.text;
+          
+          $.getJSON(get_image_for_book_uri + data.GoodreadsResponse.book.isbn13.text, function(image_object) {
+            console.info("image_object:\n");
+            console.info(image_object);
+            item['img_m'] = image_object.results[0].img_m || item['img_m'];
+          });
+        });
+      },
+      normalize: function(item) {
+        return {
+          url: "https://www.goodreads.com/book/show/" + item.best_book.id.text,
```

Comment:
nit: I would store `item.best_book` in a new variable to make these all a little less verbose:

```js
bestBook = item.best_book;
url: "..." + bestBook.id.text,
title: bestBook.title.text
...
```